src/app/task/task-list/task-list.component.ts

```diff
@@ -17,6 +17,8 @@ export class TaskListComponent implements OnInit {
   constructor(private taskService : TaskService) {}
 
   ngOnInit(): void {
-    this.taskList = this.taskService.getAllTasks();
+    this.taskService.getAllTasks().subscribe((data: any) => {
+      this.taskList = data;
```

Comment:
add 2 new methods for both event emitters (update and delete) and move functionality from task-list-item to here.

Reply:
don't forget to unsubscribe from all subscriptions